lib/system_description.rb

```diff
@@ -91,7 +91,7 @@ def validate_name(name)
     def from_hash(name, store, hash)
       begin
         json_format_version = hash["meta"]["format_version"] if hash["meta"]
-        description = SystemDescription.new(name, store, create_scopes(hash))
+        description = SystemDescription.new(name, store, create_scopes(store, name, hash))
```

Comment:
It looks wrong that `store` is used twice in the parameter list in this line. I suppose this is an indication that the code isn't structured optimally yet.